graf3d/eve7/src/REveTableProxyBuilder.cxx

```diff
@@ -8,33 +8,54 @@
 
 using namespace ROOT::Experimental;
 
+REveTableProxyBuilder::REveTableProxyBuilder() : REveDataProxyBuilderBase("Table"), fTable(0)
+{
+    fTable = new REveDataTable("ProxyTable");
+}
+
+REveTableProxyBuilder::~REveTableProxyBuilder()
+{
+}
+
+// reuse table product
+void REveTableProxyBuilder::Clean()
+{
+}
+
 void REveTableProxyBuilder::Build(const REveDataCollection* collection, REveElement* product, const REveViewContext* context)
 {
    REveTableViewInfo* info = context->GetTableViewInfo();
+   if (info->GetDisplayedCollection() != Collection()->GetElementId())
+   {
+      return;
+   }
 
-   // printf("-----REveTableProxyBuilder::Build() body for %s (%p, %p)\n",collection->GetCName(), collection, Collection() );
-   auto table = new REveDataTable("testTable");
-   table->SetCollection(collection);
-   product->AddElement(table);
-   product->SetName("product for " + collection->GetName());
-   auto tableEntries =  context->GetTableViewInfo()->RefTableEntries(collection->GetName());
-
-   for (const REveTableEntry& spec : tableEntries) {
-      auto c = new REveDataColumn(spec.fName.c_str());
-      table->AddElement(c);
-      std::string exp  = "i." + spec.fExpression + "()";
-      c->SetExpressionAndType(exp.c_str(), spec.fType);
-      c->SetPrecision(spec.fPrecision);
+   if (product->NumChildren() == 0) {
+      product->AddElement(fTable);
    }
 
-   info->SetTableId(table->GetElementId());
-   fTable = table;
+   // printf("-----REveTableProxyBuilder::Build() body for %s (%p, %p)\n",collection->GetCName(), collection, Collection() );
+
+   if (info->GetConfigChanged() || fTable->NumChildren() == 0) {
+      fTable->DestroyElements();
+      auto tableEntries =  context->GetTableViewInfo()->RefTableEntries(collection->GetName());
+      for (const REveTableEntry& spec : tableEntries) {
+         auto c = new REveDataColumn(spec.fName.c_str());
+         fTable->AddElement(c);
+         std::string exp  = "i." + spec.fExpression + "()";
```

Comment:
Does this compiles? Normally one should use explicit std::string conversion for first argument.
You can do `"i."s + spec.fExpression + "()"`, but in the begin you should have `using namespace std::string_literals;`

Reply:
Thank you. I'm learning :)